lib/csv/csv.nit

```diff
@@ -103,7 +103,7 @@ class CsvDocument
 	# * `has_header`: Is the first row the header?
 	# * `skip_empty`: Do we skip the empty lines?
 	# For details, see `CsvReader.skip_empty`.
-	fun load_from(stream: IStream, has_header: Bool, skip_empty: Bool) do
+	fun load_from(stream: Reader, has_header: Bool, skip_empty: Bool) do
```

Comment:
Actually I don't agree with you on that point, since it is a simple name it
does expose its services in a self describing way, same goes for writer, I
kind of like the fact that if your class is able to read from something it
should extend Reader as it is exactly what you intend !

Though I might be wrong on that one, StreamReader seems a bit over complex
in my opinion.
On 6 Feb 2015 18:12, "Alexandre Terrasa" <notifications@github.com> wrote:

> In lib/csv/csv.nit
> <https://github.com/privat/nit/pull/1153#discussion_r24279776>:
>
> > @@ -103,7 +103,7 @@ class CsvDocument
> >  	# * `has_header`: Is the first row the header?
> >  	# * `skip_empty`: Do we skip the empty lines?
> >  	# For details, see `CsvReader.skip_empty`.
> > -	fun load_from(stream: IStream, has_header: Bool, skip_empty: Bool) do
> > +	fun load_from(stream: Reader, has_header: Bool, skip_empty: Bool) do
>
> StreamReader?
>
> Reader seems a pretty abstract word for this case. Furthermore, I think
> simple words like this one should stay available for Nit users in their
> libs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/privat/nit/pull/1153/files#r24279776>.
>